python/ml4ir/applications/classification/model/classification_model.py

```diff
@@ -62,52 +63,69 @@ def evaluate(
         -----
         You can directly do a `model.evaluate()` only if the keras model is compiled
         """
+        # FIXME: We need to come to a consensus for computing in pandas or spark and how we plan to
+        #  do that. In the mean time we will stick with just the model evaluation metrics
+        compute_group_metrics_in_memory = False
+
         if not self.is_compiled:
             return NotImplementedError
         group_metrics_keys = self.feature_config.get_group_metrics_keys()
         metrics_dict = self.model.evaluate(test_dataset)
```

Comment:
Quick callout: This works fine for 7 days of data without issues (on a 40 gig container). It's just extremely slow as one would expect. (A potential solution if this is causing issues later is to increase the batch size for test significantly as inference is less memory intensive than training)

Reply:
yes. Also we are talking about 40M data points here.. I am not too in favour of increasing the batch size under the hood.
how much time does it take?

Reply:
I didn't mean to say we should change it under the hood. Should be changed in the arguments. It takes well over 4 hours